### PR TITLE
Data type of custom claim for OAuth2

### PR DESCRIPTION
**Pull request by yuliu-okta**
OKTA-113034

Reviewer: @larsjohansen-okta @richardmateosian-okta @federations-okta

### CODE REVIEW

**Comment by larsjohansen-okta** [comment]
🚢

File: _source/_docs/api/resources/oauth2.md [hunk 1]
```diff
@@ -183,7 +187,7 @@ If the request that generates the access token contains any custom scopes, those
 
 #### Custom claims
 
-Custom claims are associated with scopes. If one of the associated scopes is granted to the access token, the custom claim will be added into it. The value of a custom claim can be either an [expression](/reference/okta_expression_language) or group filter. The expression will be evaluated at runtime, and if the evaluated result is null, that custom claim will not be added into the access token.
+Custom claims are associated with scopes. If one of the associated scopes is granted to the access token, the custom claim will be added into it. The value of a custom claim can be either an [expression](/reference/okta_expression_language) or group filter. The expression will be evaluated at runtime, and if the evaluated result is null, that custom claim will not be added into the access token. The data type of a claim is Array if its value is a group filter, or the same data type as the evaluated result if its value is an expression.
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: A few changes (the doc is inconsistent right now access token vs. Access Token, but Karl/Trevor asked us to use the spec conventions):

Custom claims are associated with scopes. If one of the associated scopes is granted to the Access Token, the custom claim is added into it. The value of a custom claim can be either an [expression](/reference/okta_expression_language) or a group filter. The expression is evaluated at runtime, and if the evaluated result is null, that custom claim isn't added into the Access Token. The datatype of a claim is Array if its value is a group filter, or the same datatype as the evaluated result if its value is an expression.

Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: I also changed a few verbs to simple present and data type to datatype. You can just copy the whole paragraph if you want.

File: _source/_docs/api/resources/oauth2.md [hunk 2]
```diff
@@ -441,7 +445,7 @@ http://www.example.com/#error=invalid_scope&error_description=The+requested+scop
 
 {% api_operation post /oauth2/:authorizationServerId/v1/token %}
 
-The API takes a grant type of either *authorization_code*, *password*, *refresh_token*, or [*client_credentials* {% api_lifecycle beta %}](/docs/api/getting_started/releases-at-okta.html) and the corresponding credentials and returns back an Access Token. A Refresh Token will be returned if *offline_access* scope is requested using authorization_code, password, or refresh_token grant type. Additionally, using the authorization_code grant type will return an ID Token if the *openid* scope is requested.
+The API takes a grant type of either *authorization_code*, *password*, *refresh_token*, or *client_credentials* and the corresponding credentials and returns back an Access Token. A Refresh Token will be returned if *offline_access* scope is requested using authorization_code, password, or refresh_token grant type. Additionally, using the authorization_code grant type will return an ID Token if the *openid* scope is requested.
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: change "will be" and "will return" to simply "is" and "returns" (although technically correct, in tech doc we try to stick to simple present tense unless it's needed for clarity.

File: _source/_docs/api/resources/oauth2.md [hunk 3]
```diff
@@ -451,7 +455,7 @@ The following parameters can be posted as a part of the URL-encoded form values
 
 Parameter          | Description                                                                                         | Type       |
 -------------------+-----------------------------------------------------------------------------------------------------+------------|
-grant_type         | Can be one of the following: *authorization_code*, *password*, *refresh_token*, or [*client_credentials* {% api_lifecycle beta %}](/docs/api/getting_started/releases-at-okta.html). Determines the mechanism Okta will use to authorize the creation of the tokens. | String |  
+grant_type         | Can be one of the following: *authorization_code*, *password*, *refresh_token*, or *client_credentials*. Determines the mechanism Okta will use to authorize the creation of the tokens. | String |
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: "will use" to "uses"

File: _source/_docs/api/resources/oauth2.md [hunk 4]
```diff
@@ -612,7 +614,7 @@ invalid_request         | The request structure was invalid. E.g. the basic auth
 {
     "active" : true,
     "token_type" : "Bearer",
-    "scope" : "openid profile",
+    "scope" : "openid email flights custom",
```
Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: Did we doc the flights response param anywhere?

File: _source/_docs/api/resources/oauth2.md (hunk 1 above)
Comment (yuliu-okta) [inline _source/_docs/api/resources/oauth2.md]: Fixed, thanks for writing the whole paragraph for me :)

File: _source/_docs/api/resources/oauth2.md (hunk 2 above)
Comment (yuliu-okta) [inline _source/_docs/api/resources/oauth2.md]: fixed

File: _source/_docs/api/resources/oauth2.md (hunk 3 above)
Comment (yuliu-okta) [inline _source/_docs/api/resources/oauth2.md]: fixed

File: _source/_docs/api/resources/oauth2.md (hunk 4 above)
Comment (yuliu-okta) [inline _source/_docs/api/resources/oauth2.md]: It is just an example of custom scope, it is not a reserved one or a response param.

Comment (mystiberry-okta) [inline _source/_docs/api/resources/oauth2.md]: Perfect. 🚢  Is it safe to merge now or should I wait for 2017.05?

Comment (yuliu-okta) [inline _source/_docs/api/resources/oauth2.md]: Please wait for 2017.05.

**Pull request merged**